File: leetcode_solution/Remove Duplicates from Sorted List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    # @param head, a ListNode
    # @return a ListNode
    def deleteDuplicates(self, head):
		if head == None:
			return None
		if head.next == None:
			return head	
		another = head
		while head.next != None: 
		    while head.val != head.next.val:
		        head.next = head.next
                head = head.next
            head.next = None
		return another
#Error_1 Time Limit Exceeded:
#

File: leetcode_java/Remove Duplicates from Sorted List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
// public class Solution {
//     public ListNode deleteDuplicates(ListNode head) {
        
//     }
// }
public class Solution {
    public ListNode deleteDuplicates(ListNode head) {

		if (head == null || head.next == null) {
			return head;
		} else {
			ListNode headForward = head;
			ListNode compareNode = head.next;

			while (compareNode != null) {
				if (compareNode.val == headForward.val) {
					headForward.next = compareNode;
					compareNode = compareNode.next;
				} else {
					headForward =  compareNode.next;
					compareNode = compareNode.next;
				}
			}
			return head;
		}
    }
}